ch11.md

```diff
@@ -0,0 +1,225 @@
+
+## One nesting solution
+
+Back to our comedic type signature. We can sprinkle in some *natural transformations* throughout the calling code to coerce each varying type so they are uniform and, therefore, `join`able.
+
+```js
+// getValue :: Selector -> Task Error (Maybe String)
+// postComment :: String -> Task Error Comment
+// validate :: String -> Either ValidationError String
+
+// saveComment :: () -> Task Error Comment
+const saveComment = compose(chain(postComment), chain(eitherToTask), map(validate), chain(maybeToTask), getValue('#comment'))
+```
+
+So what do we have here? We've simply added `chain(maybeToTask)` and `chain(eitherToTask)`. Both have the same effect; they naturally transform the functor our `Task` is holding into another `Task` then `join` the two. Like pigeon spikes on a window ledge, we avoid nesting right at the source. As they say in the city of light, "Mieux vaut prévenir que guérir" - an ounce of prevention is worth a pound of cure.
+
+## In summary
+
+*Natural transformations* are functions on our functors themselves. They are an extremely important concept in category theory and will start to appear everywhere once more abstractions are adopted, but for now, we've scoped them to a few concrete applications. As we saw, we can achieve different effects by converting types with the guarantee that our composition will hold. They can also help us with nested types, although they have the general effect of homogenizing our functors to the lowest common denominator, which in practice, is the functor with the most volatile effects (`Task` in most cases).
+
+This continual and tedious sorting of types is the price we pay for having materialized them - summoned them from the ether. Of course, implicit effects are much more insidious and so here we are fighting the good fight. We'll need a few more tools in our tackle before we can reel in the larger type amalgamations. Next up, we'll look at reordering our types with *Traversable*.
+
+[Chapter 12: Traversing the Stone](ch12.md)
+
+
+## Exercises
+
+```js
+require('../../../part2_exercises/support');
+const {map, compose, prop, chain, sortBy, identity, split, join} = require('ramda');
+const Task = require('data.task');
+
+// Exercise 1
+// ==========
+// Write a natural transformation that converts `Either b a` to `Maybe a`
+
+//  ex1 :: Either b a -> Maybe a
+const ex1 = identity // write me
+
+
+// Exercise 2
+// ==========
+// Use the eitherToTask natural transformation change ex2's type signature
+// from :: Number -> Task Error (Either Error String)
+// to :: Number -> Task Error String
+
+// findUser :: Number -> Task Error (Either Error User)
+const findUser = x => x > 0 ? Task.of(Either.of({id: x, name: 'userface'})) : Task.of(new Left('not found'))
+
+// eitherToTask :: Either a b -> Task a b
+const eitherToTask = either(Task.rejected, Task.of)
```

Comment:
`either` is not defined.
Did you want to bring it in or should it have been created with exercise 1?

Reply:
It's part of support.js and we use it in the exercises somewhere. I think we should just add the type signature in a comment at the top or something...

I can't believe how "old school" the module system with globals and such are in the code itself. Might be a better way to actually link to the function from here.

Reply:
This is going to change with the upcoming restructuring update :+1: